### ISSUE TITLE
strip symbols on windows and mac

### ISSUE THREAD

**Issue by scopatz** [regro/conda-press#6]
We need to be able to reduce the binary size on platforms such as Windows and Mac by stripping symbols out of dlls, dylibs, executables, etc. 

I have pretty much no idea how to do this, but the implementation would go into this method: https://github.com/regro/conda-press/blob/master/conda_press/condatools.xsh#L589

CC @matthew-brett @stefanv

**Comment by UchihaSR** [comment]
I am interested in this project, I would like to contribute to it.  would you elaborate on this issue for me (in layman's term)

**Comment by scopatz** [comment]
Hi @UchihaSR! Thanks for the interest, and we'd love to have your help!  This issue might not be the best one to start out with...

However, the basic issue is the share object libraries (.so, .dylib, .lib) files contain a lot of extra "symbol" information in them usually that isn't needed for the running of the code. For `*.so` files, the `strip` command line utility will get rid of thes extra information. (`strip` usually part of the `binutils` package.) This can dramatically reduce the size of the `.so` files so that certain wheels I was testing went from 200 MB to 40 MB.  

I believe that there are equivalent tools to `strip` for `.dylib` and `.lib` files, but I don't know what they are or how to use them.  This issue is to get that integration.
